### PR TITLE
fix dialect(::Operation)

### PR DESCRIPTION
**Pull request by Pangoraw**

### CODE REVIEW

**Comment by Pangoraw** [comment]
I am not sure about the Symbol return type for this function, maybe it should be String?

**Comment by vchuravy** [comment]
> I am not sure about the Symbol return type for this function, maybe it should be String?

I think either would be fine

File: test/ir.jl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using MLIR.Dialects: arith
+using MLIR.IR, LLVM
+
+@testset "operation introspection" begin
+    IR.context!(IR.Context()) do
+        IR.get_or_load_dialect!("linalg")
```
Comment (vchuravy) [inline test/ir.jl:6]: Should this have been `arith`?

Comment (Pangoraw) [reply test/ir.jl:6]: unfortunately, `mlirGetDialectHandle__arith__` does not exist so I had to load transitively with linalg.

**Pull request merged**